### PR TITLE
Apache Pulsar

### PR DESCRIPTION
**Pull request by MickeyPvX**
Related: https://github.com/chainguard-dev/image-requests/issues/1654

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

File: apache-pulsar.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  - name: ${{package.name}}-compat
+    pipeline:
+      - runs: ln -s /usr/share/java/pulsar ${{targets.contextdir}}/pulsar
+
+update:
+  enabled: true
+  github:
+    identifier: apache/pulsar
+    use-tag: true
+    strip-prefix: v
```
Comment (jamonation) [inline apache-pulsar.yaml:50]: There'll need to be a tag filter here to ignore things like candidate versions, e.g. https://github.com/apache/pulsar/releases/tag/v4.0.1-candidate-2

File: apache-pulsar.yaml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+test:
+  environment:
+    contents:
+      packages:
+        - ${{package.name}}-compat
+        - bash
+        - libstdc++
```
Comment (EyeCantCU) [inline apache-pulsar.yaml:58]: If these are needed by the package, let's add them as runtime deps

File: apache-pulsar.yaml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        - openjdk-17-default-jvm
+        - openjdk-17-jre
```
Comment (EyeCantCU) [inline apache-pulsar.yaml:60]: Fetched by the package

File: apache-pulsar.yaml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  name: apache-pulsar
+  version: 4.0.1
+  epoch: 0
+  description: Pulsar is a distributed pub-sub messaging platform with a very flexible messaging model and an intuitive client API.
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - openjdk-17-default-jvm
+      - openjdk-17-jre
```
Comment (EyeCantCU) [inline apache-pulsar.yaml:11]: `-default-jvm` will pull in `-jre`, we should probably use `-jre` and install `-default-jvm` at the image level

File: apache-pulsar.yaml (hunk 2 above)
Comment (MickeyPvX) [reply apache-pulsar.yaml:58]: I figured out what I was doing wrong (or maybe not-quite-right?) - I'd been testing the package build with `make dev-container` but I think that has some extra packages on board that would make me miss some of the runtime dependencies.

Re-read the Building Packages docs, switched to testing with `make local-wolfi` and that actually replicates all the runtime failures I was seeing when doing `make test/apache-pulsar`

File: apache-pulsar.yaml (hunk 4 above)
Comment (MickeyPvX) [reply apache-pulsar.yaml:11]: So I took out the `-default-jvm` leaving just the `-jre` and I get the common `Error: JAVA_HOME not set, and no java executable found in /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin` - I assume that's expected now until we install the `-default-jvm` in the image or specify it as part of what we need for testing?

**Review by jamonation — DISMISSED** [review]

File: apache-pulsar.yaml (hunk 4 above)
Comment (EyeCantCU) [reply apache-pulsar.yaml]: Yep, you'd either export `JAVA_HOME` or install `-default-jvm`

**Review by EyeCantCU — APPROVED** [review]

**Review by jamonation — APPROVED** [review]

**Pull request merged**